plugins/filter/merge_data.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022     , DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+name: merge_data
+short_description: Compare one data structure provided in I(_input) with an
+                   other provided in I(expected) and ensure that these data are
+                   presents or absents in the result depending the I(state).
+version_added: 5.2.0
+author: DEMAREST Maxime (@indelog)
+description:
+  - Compare data structure provided in I(_input) with data structure provided
+    in I(expected).
+positional: expected
+options:
+  _input:
+    description:
+      - The current data structure that you want to compare with I(expected).
+      - This can be one of a dictionary or a list that can contain itself
+        other dictionaries and lists.
+    type: raw
+    required: true
+extends_documentation_fragment: community.general.merge_data
+ '''
+
+EXAMPLES = r'''
+- name: Ensure that the expected data will be present with lists merged by values.
+  ansible.builtin.debug:
+    msg: "{{ current|community.general.merge_data(expected, state='present') }}"
```

Comment:
The `state` parameter feels really strange for a filter. I would rather expect an operation here, as a filter doesn't ensure idempotency, but modifies something. Especially the value `identic` seems wrong, doesn't it simply mean "return `expected`"?